spec/PasswordPolicy.spec.js

```diff
@@ -0,0 +1,697 @@
+"use strict";
+
+const request = require('request');
+const Config = require('../src/Config');
+
+describe("Password Policy: ", () => {
+
+  it('should show the invalid link page if the user clicks on the password reset link after the token expires', done => {
+    const user = new Parse.User();
+    let sendEmailOptions;
+    const emailAdapter = {
+      sendVerificationEmail: () => Promise.resolve(),
+      sendPasswordResetEmail: options => {
+        sendEmailOptions = options;
+      },
+      sendMail: () => {
+      }
+    }
+    reconfigureServer({
+      appName: 'passwordPolicy',
+      emailAdapter: emailAdapter,
+      passwordPolicy: {
+        resetTokenValidityDuration: 0.5, // 0.5 second
+      },
+      publicServerURL: "http://localhost:8378/1"
+    }).then(() => {
+      user.setUsername("testResetTokenValidity");
+      user.setPassword("original");
+      user.set('email', 'user@parse.com');
+      return user.signUp();
+    })
+      .then(user => {
+        Parse.User.requestPasswordReset("user@parse.com");
+      })
+      .then(() => {
+        // wait for a bit more than the validity duration set
+        setTimeout(() => {
+          expect(sendEmailOptions).not.toBeUndefined();
+
+          request.get(sendEmailOptions.link, {
```

Comment:
@bhaskaryasa 
request-promise provides a nice promise way of calling get. 
Can u pls look into using it?